airbyte-integrations/connectors/destination-snowflake/src/main/kotlin/io/airbyte/integrations/destination/snowflake/typing_deduping/SnowflakeDestinationHandler.kt

```diff
@@ -70,75 +71,83 @@ class SnowflakeDestinationHandler(
-            |FROM information_schema.tables
-            |WHERE table_catalog = ? 
-            |AND table_schema IN (${IntRange(1, streamIds.size).joinToString { "?" }}) 
-            |AND table_name IN (${IntRange(1, streamIds.size).joinToString { "?" }})
-            |""".trimMargin()
-        val bindValues = arrayOf(databaseName) + namespaces + names
-        val results: List<JsonNode> = database.queryJsons(query, *bindValues)
-        for (result in results) {
-            val tableSchema = result["TABLE_SCHEMA"].asText()
-            val tableName = result["TABLE_NAME"].asText()
-            val rowCount = result["ROW_COUNT"].asInt()
-            tableRowCounts
-                .computeIfAbsent(tableSchema) { _: String? -> LinkedHashMap() }[tableName] =
-                rowCount
+
+        val tableRowCountsFromShowQuery = LinkedHashMap<String, LinkedHashMap<String, Int>>()
+        var showColumnsResult: List<JsonNode> = listOf()
+
+        try {
+            for (stream in streamIds) {
+                val showColumnsQuery =
+                    String.format(
+                            """
+                                SHOW TABLES LIKE '%s' IN "%s"."%s";
+                            """.trimIndent(),
+                            stream.finalName,
+                            databaseName,
+                            stream.finalNamespace,
+                            )
+                showColumnsResult = database.queryJsons(
+                    showColumnsQuery,
+                )
+                for (result in showColumnsResult) {
+                    val tableSchema = result["schema_name"].asText()
+                    val tableName = result["name"].asText()
+                    val rowCount = result["rows"].asText()
+
+                    tableRowCountsFromShowQuery
+                        .computeIfAbsent(tableSchema) { _: String? -> LinkedHashMap() }[tableName] =
+                        rowCount.toInt()
+                }
+            }
+        } catch (e: SQLException) {
+            showColumnsResult.stream().close()
+            //Not re-throwing the exception since the SQLException occurs when the table does not exist
+            //throw e
         }
-        return tableRowCounts
+        return tableRowCountsFromShowQuery
     }
 
+
     @Throws(Exception::class)
     private fun getInitialRawTableState(
         id: StreamId,
         suffix: String,
     ): InitialRawTableStatus {
+
         val rawTableName = id.rawName + suffix
-        val tableExists =
-            database.executeMetadataQuery { databaseMetaData: DatabaseMetaData ->
-                LOGGER.info(
-                    "Retrieving table from Db metadata: {} {}",
+        var tableExists = false
+        var showTablesResult: List<JsonNode> = listOf()
+
+        try {
+            val showTablesQuery =
+                String.format(
+                    """
+                        SHOW TABLES LIKE '%s' IN "%s"."%s";
```

Comment:
use kotlin templates instead of `String.format`

Reply:
does that work when the QUOTED_IDENTIFIERS_IGNORE_CASE is set to true? We have a test class that sets that for the testDatabase